### PR TITLE
Revise service management end to end workflow documentation

### LINKED ISSUE #906: SPK install-build-pipeline issue around location of azure-pipelines.yaml file
**Describe the bug:**

This [line](https://github.com/CatalystCode/spk/blob/93ddacdc77aa9513ff49e89f92d1ef00280050b0/src/commands/service/pipeline.ts#L213) seem to assume that in a non-mono repo we would still have a `services` directory instead of just having `azure-pipeline.yaml` exist at the top level of the repository.

**To Reproduce:**

When you have a single application repository (not a mono-repo)

`spk service install-build-pipeline -o abrig -u https://dev.azure.com/abrig/spk/_git/fabrikam-frontend -d spk -p REDACTED -n fabrikam.frontend.pipeline fabrikam-frontend`

```bash
warn:    Error loading SPK configuration file; run `spk init` to initialize your global configuration or ensure you have passed all required parameters to the called function.
(node:4624) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
info:    Fetched DevOps Client
info:    Creating pipeline for definition
info:    Created pipeline for fabrikam.frontend.pipeline
info:    Pipeline ID: 416
error:   Could not queue the build because there were validation errors or warnings.
error:   Error occurred when queueing build for fabrikam.frontend.pipeline
error:   Error queueing build
```

**This happens with the latest master build of SPK**

The build cannot be queued because AzDO cannot find the `azure-pipeline.yaml` file. The `azure-pipeline.yaml` file exists at the top level of the repo, **not** under the directory `fabrikam-frontend`. The image below shows the issue:

image

**Expected behavior:**
SPK allow build pipelines to be created with a `azure-pipeline.yaml` file living at the top level of a single application repository

### PR DESCRIPTION
**Pull request by timfpark**
Also closes https://github.com/microsoft/bedrock/issues/906

### CODE REVIEW

File: docs/project-service-management-guide.md [hunk 1]
```diff
@@ -156,106 +152,101 @@ deploy all manifests in this repository to the cluster in a set interval.
...
+#### Adding a Service to a Application Repository
 
 - [Clone the repository.](https://docs.microsoft.com/en-us/azure/devops/repos/git/create-new-repo?view=azure-devops#clone-the-repo-to-your-computer)
 - Create the service via `spk`, there are optional parameters that _should_ be
   used to configure the service and its associated helm charts (other optional
   flag parameters can be used if `spk` was not intialized)
   ```
   SERVICE_NAME=<my-new-service-name>
-  spk service create $SERVICE_NAME
+  spk service create . --display-name $SERVICE_NAME ...
```
Comment (mtarng) [inline docs/project-service-management-guide.md:233]: I kind of want to steer users away from using the `.` notation for placing their services under the root directory, or perhaps we can have explicit instructions for placing a service in the project root vs. placing the service in a subdirectory.

Do we envision more single application repositories or "monorepos"?

Hopefully our refactoring will make the behaviors clear and consistent soon.

File: docs/project-service-management-guide.md [hunk 2]
```diff
@@ -156,106 +152,101 @@ deploy all manifests in this repository to the cluster in a set interval.
...
   where `ACR_NAME` is the name of the Azure Container Registry where the Docker
   Images will be served from and `SP_APP_ID`, `SP_PASS`, and, `SP_TENANT` are an
   associated Service Principal's ID, Password, and Tenant, that have Read and
   Write access to the ACR.
 - Deploy the lifecycle pipeline (optional flag parameters can be used if `spk`
   was not intialized)
+
   ```
-  spk project install-lifecycle-pipeline
+  $ spk project install-lifecycle-pipeline --org-name $ORG_NAME --devops-project $DEVOPS_PROJECT --repo-url $SERVICE_REPO_URL --repo-name $SERVICE_NAME --pipeline-name $SERVICE_NAME-lifecycle-pipeline
```
Comment (mtarng) [inline docs/project-service-management-guide.md]: `--pipeline-name $SERVICE_NAME-lifecycle-pipeline`

isn't strictly necessary, the default behavior would be <projectreponame>-lifecycle-pipeline .

**Comment by mtarng** [comment]
Thanks for streamlining this guide!

File: docs/project-service-management-guide.md (hunk 1 above)
Comment (timfpark) [reply docs/project-service-management-guide.md:233]: I think we will have a mix of both monorepos and single repo per service so we should describe both.

File: docs/project-service-management-guide.md (hunk 2 above)
Comment (timfpark) [reply docs/project-service-management-guide.md]: Ok I will drop it then.

**Review by mtarng — APPROVED** [review]

**Review by edaena — APPROVED** [review]

**Pull request merged**